### PR TITLE
Move firstName,lastName to MulitUser

### LINKED ISSUE #9105: Move user firstName, lastName to multiUsers table
As discussed in https://scm.slack.com/archives/C5AKLAV0B/p1764149494650819?thread_ts=1764071136.741679&cid=C5AKLAV0B

The evolution should just take the value of the first matching user. We can check manually first if any users have ambiguous names and fix that where needed.

### PR DESCRIPTION
**Pull request by fm3**
All users belonging to a MultiUser already had to have the same name. The logical cleanup is to move these fields to the MultiUser directly.

Also fixed gitignore for type_check/ files, as mentioned in https://scm.slack.com/archives/C5AKLAV0B/p1773329610452609

### URL of deployed dev instance (used for testing):
- https://multiusername.webknossos.xyz

### Steps to test:
- Annotate some, download, user name should be in NML (and abbreviated in filename unless explicitly named via UI)
- Run vx jobs (e.g. via worker), workflow list should show correct user name
- Users should still be able to change their own name
- User names should still be in annotation list view

### Issues:
- fixes #9105

------
- [x] Added migration guide entry if applicable (edit the same file as for the changelog)
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)

### CODE REVIEW

File: frontend/javascripts/test/snapshots/type_check/test_type_checking_activeUser.ts
Comment (MichaelBuessemeyer) [inline frontend/javascripts/test/snapshots/type_check/test_type_checking_activeUser.ts:1]: Are these files present on the master 🤔 ? Thanks for deleting them btw 🎉

File: schema/schema.sql [hunk 1]
```diff
@@ -502,6 +500,8 @@ CREATE TABLE webknossos.multiUsers(
   email TEXT NOT NULL UNIQUE CHECK (email ~* '^.+@.+$'),
   passwordInfo_hasher webknossos.USER_PASSWORDINFO_HASHERS NOT NULL DEFAULT 'SCrypt',
   passwordInfo_password TEXT NOT NULL,
+  firstName TEXT NOT NULL, -- CHECK (firstName ~* '^[A-Za-z0-9\-_ ]+$'),
+  lastName TEXT NOT NULL, -- CHECK (lastName ~* '^[A-Za-z0-9\-_ ]+$'),
```
Comment (MichaelBuessemeyer) [inline schema/schema.sql]: Why is the CHECK statement commented out here? I know this was the case previously as well but why not add it? Are there names contradicting this?

File: app/models/user/MultiUser.scala [hunk 2]
```diff
@@ -63,17 +72,19 @@ class MultiUserDAO @Inject()(sqlClient: SqlClient)(implicit ec: ExecutionContext
       )
     }
 
-  def insertOne(u: MultiUser): Fox[Unit] =
+  def insertOne(mu: MultiUser): Fox[Unit] =
```
Comment (MichaelBuessemeyer) [inline app/models/user/MultiUser.scala:75]: 🐮

File: app/models/user/User.scala [hunk 3]
```diff
@@ -287,8 +278,8 @@ class UserDAO @Inject()(sqlClient: SqlClient)(implicit ec: ExecutionContext)
             u._id,
             m._id,
             m.email,
-            u.firstName,
-            u.lastName,
+            m.firstName,
```
Comment (MichaelBuessemeyer) [inline app/models/user/User.scala:281]: just a thought: In the MultiUser.scala you used `mu` as a var name in the queries. Maybe use this here as well 🤔?

File: app/models/user/MultiUser.scala [hunk 4]
```diff
@@ -153,6 +172,19 @@ class MultiUserDAO @Inject()(sqlClient: SqlClient)(implicit ec: ExecutionContext
...
+  def findMultiUserOfOrganizationOwner(organizationId: String): Fox[MultiUser] =
+    for {
+      r <- run(q"""SELECT ${columnsWithPrefix("mu")}
+                   FROM webknossos.users_ u
+                   JOIN webknossos.multiUsers_ mu ON u._multiUser = mu._id
+                   WHERE u.isOrganizationOwner
+                   AND NOT u.isDeactivated
+                   AND u._organization = $organizationId
+                   ORDER BY mu._id
+                   LIMIT 1""".as[MultiusersRow])
```
Comment (MichaelBuessemeyer) [inline app/models/user/MultiUser.scala]: Trying to go into 🤓 mode:
Isn't the order by unnecessary? Shouldn't there always be one owner at a time? And shouldn't the limit 1 ensure that there is only one user returned anyway?

Or do you want to guard against the case where accidentally multiple users are owners and that this call consistently returns the same user in that case 🤔 ?

Comment (MichaelBuessemeyer) [reply app/models/user/MultiUser.scala]: If my later assumption is correct, could you maybe please add a comment about this?

File: frontend/javascripts/test/snapshots/type_check/test_type_checking_activeUser.ts
Comment (fm3) [reply frontend/javascripts/test/snapshots/type_check/test_type_checking_activeUser.ts:1]: yes, compare https://scm.slack.com/archives/C5AKLAV0B/p1773329869114549?thread_ts=1773329610.452609&cid=C5AKLAV0B

File: schema/schema.sql (hunk 1 above)
Comment (fm3) [reply schema/schema.sql]: I’ll remove the commented out lines, they were leftovers on master. Indeed, there are users with names that don’t fit this, and I think they should keep their names.

File: app/models/user/MultiUser.scala (hunk 4 above)
Comment (fm3) [reply app/models/user/MultiUser.scala]: I’ll remove the order by
